crates/sui-source-validation-service/src/main.rs

```diff
@@ -29,13 +30,39 @@ pub async fn main() -> anyhow::Result<()> {
     info!("verification complete in {:?}", start.elapsed());
 
     let app_state = Arc::new(RwLock::new(AppState { sources }));
-    let app_state_copy = app_state.clone();
     let mut threads = vec![];
-    let watcher =
-        tokio::spawn(
-            async move { watch_for_upgrades(package_config.packages, app_state, None).await },
-        );
-    threads.push(watcher);
+    let networks_to_watch = vec![
+        Network::Mainnet,
+        Network::Testnet,
+        Network::Devnet,
+        Network::Localnet,
+    ];
+    // spawn a watcher thread for upgrades for each network
+    for network in networks_to_watch {
+        let app_state_copy = app_state.clone();
+        let packages: Vec<_> = package_config
+            .clone()
+            .packages
+            .into_iter()
+            .filter(|p| match p {
+                PackageSource::Repository(RepositorySource {
+                    network: Some(n), ..
+                })
+                | PackageSource::Directory(DirectorySource {
+                    network: Some(n), ..
+                }) => *n == network,
+                _ => false,
+            })
```

Comment:
nit: This might be a candidate for `matches!`

```suggestion
            .filter(|p| matches!(
                p, 
                PackageSource::Repository(RepositorySource {
                    network: Some(n), ..
                })
                | PackageSource::Directory(DirectorySource {
                    network: Some(n), ..
                }) if *n == network,
            })
```

Reply:
I missed (forgot?) about `matches!`, done, ty!